apps/emqx/src/emqx_broker.erl

```diff
@@ -94,6 +103,17 @@
     end
 ).
 
+%% Subscription scope.
+%% When incoming message matches the subscription topic filter, scope can additionally
+%% narrow down message delivery.
+%% * Scope `any` means a subscription with this scope is interested in all messages.
+%% * Scope `qos0` means a subscription needs _only_ QoS 0 messages.
+%%   This is currently employed by durable sessions to route QoS 0 messages through
+%%   the realtime channel to avoid storing them in the durable storage.
+%% Currently, this affects only the dispatching phase, but not the routing phase.
+%% Messages are routed between nodes regardless of which subscription scopes nodes have.
+-type subscope() :: any | qos0.
```

Comment:
Suggestion, maybe expose something like this instead:

`-type channel() :: direct | direct_0.`

...to reduce the number of terms we have to juggle around when reasoning about different ways to forward messages between the clients from the high-level perspective.

Instead of speaking in terms of "durable" vs. "realtime" (or "direct") with scope any or scope qos0, we can just say we have three "logical channels":

- direct
- durable
- direct_0

Reply:
I've read the proposal but I think that _channel_ is:
* Already has an established meaning in the `emqx` app, there's `emqx_channel`, `emqx_cm`, etc.
* Something from the communication domain, but in reality communication is happening through the same mechanisms as before.
* Does not map very well to the routing context, upon which most of this functionality is built.

Because of these I decided to go with something closer to the routing context and build on top of them.

On the other hand, agree with `direct` vs `realtime`, will do quick refactoring.